### PR TITLE
Make remaining pages translation-ready

### LINKED ISSUE #419: Some pages aren't translation-ready
From #413:

> When looking at @stardust66's PR over at [Intimaria/refugerestrooms:#1](https://github.com/Intimaria/refugerestrooms/pull/1), I noticed that we have some pages that aren't translation-ready.
> 
> We need to update:
> - the "splash" page (homepage),
> - the Contact page,
> - the search results page,
> - the nav bar (at the top)
> - the footer
> - search bar buttons
> - ARIA labels (probably)
> - and if somehow possible, the API documentation page. 
> 
> Making them "translation-ready" would involve taking "hard-coded" sentences and phrases and moving them into a `[something].en.yaml` file, then updating the actual pages to refer to translation strings in place of hard-coded phrases. In other words, expand the existing translation system to also cover the pages I mentioned.

Solving this helps us to progress with #396 and ensure we can more-comprehensively complete #413

### PR DESCRIPTION
**Pull request by DeeDeeG**
# Context
- Fixes #419 
- Replaces "hard-coded" English text with translation strings
  - ~~Only splash page is done as of writing, but will be adding more commits soon.~~
  - All strings I could find are translated.
  - Potentially open to collaboration if we announce which pages we all are working on ahead of time.
# Summary of Changes

- Changes `html.haml` files to use translation strings
- Adds the translation strings in `config/locales/[component_name].en.yml`

# Checklist

- [x] Tested Mobile Responsiveness
- [ ] Added Unit Tests
- [x] CI Passes
- [ ] Deploys to Heroku on test Correctly (Maintainers will handle)
- [ ] Added Documentation (Service and Code when required)
---

> We need to update:
> - [x] the "splash" page (homepage),
> - [x] the Contact page,
>   - [x] the "Thank you for contacting us" page ~~_(can't figure this one out, skipping)_~~
> - [x] simple_form content (HTML input forms, labels and such)
> - [x] the search results page and restroom entries,
> - [x] the nav bar (at the top)
> - [x] the footer
> - [x] search bar buttons
> - [x] ARIA labels
> - [ ] and if somehow possible, the API documentation page. _(Does not seem doable IMO)_

### CODE REVIEW

**Comment by DeeDeeG** [comment]
This is ready to be looked over/reviewed. @tkwidmer, @mi-wood, mind taking a look at this?

File: app/views/contacts/new.html.haml [hunk 1]
```diff
@@ -1,7 +1,12 @@
 - if @contact.restroom_name.present?
-  %h1 Request Edit for #{@contact.restroom_name} Restroom
+  - titlestring = t('.request-edit-for')
```
Comment (tkwidmer) [inline app/views/contacts/new.html.haml]: You can actually interpolate this. `t('request-edit-for', restroom_name: @contact.restroom_name)` and then define the translation like this:
`request-edit-for: 'Request Edit for %{restroom_name} Restroom'`

Comment (tkwidmer) [reply app/views/contacts/new.html.haml]: I18n, takes the param restroom_name and shoves it into the %{restroom_name} param

File: app/views/contacts/new.html.haml [hunk 2]
```diff
@@ -10,6 +15,6 @@
   .hidden
     = f.hidden_field :restroom_id, :required => false
     = f.hidden_field :restroom_name, :required => false
-    = f.hidden_field :nickname, :hint => 'Leave this field blank!'
+    = f.hidden_field :nickname, :hint => t('.leave-this-field-blank')
   .form-actions
-    = f.button :submit, 'Send message', :class=> "linkbutton"
```
Comment (tkwidmer) [inline app/views/contacts/new.html.haml]: I don't see a translation here?

**Comment by DeeDeeG** [comment]
Thanks for the comments, will get to these as soon as possible.

File: app/views/contacts/new.html.haml (hunk 2 above)
Comment (DeeDeeG) [reply app/views/contacts/new.html.haml]: It was a weird experience trying to translate this, as buttons in `simple_form` are apparently from some other dimension compared to all other parts of forms, and try as I might, I couldn't figure out a way to translate this with the usual `t('key.name.here')` format, inline in `app/views/contacts/new.html.haml`.

The translation ended up here: [simple_form.en.yml, Line 27](https://github.com/RefugeRestrooms/refugerestrooms/pull/421/files#diff-596374a12ef429714137455430e6b4c4R27)

This seemed like the logical place to put it.

I do notice that I accidentally changed the string from "_Send_ message" -> "_Submit_ message" so I can change that back if you like.

**Comment by DeeDeeG** [comment]
This is ready for another round of reviews, when someone gets the chance. @tkwidmer, @mi-wood

**Comment by DeeDeeG** [comment]
I am working on translating the "Thanks for contacting us" strings. They're in [`app/controllers/contacts_controller.rb`](https://github.com/refugerestrooms/refugerestrooms/blob/develop/app/controllers/contacts_controller.rb)

**Comment by DeeDeeG** [comment]
Google Analytics broke one of the tests. :^/

```
  Scenario: Show nearby restrooms when guessing                          # features/nearby.feature:4
    Given a restroom exists in Winnipeg                                  # features/step_definitions/restroom_steps.rb:1
Started GET "/" for 127.0.0.1 at 2018-01-19 19:37:28 +0000
Processing by PagesController#index as HTML
  Rendering pages/index.html.haml within layouts/splash
  Rendered layouts/_search.html.haml (6.6ms)
 [ Skipping a few lines for brevity . . . ]
    When I am in Winnipeg and I guess my location on the submission page # features/step_definitions/submission_steps.rb:29
      Request to 'http://127.0.0.1:36316/' failed to reach server, check DNS and/or server status - Timed out with the following resources still waiting http://www.google-analytics.com/analytics.js (Capybara::Poltergeist::StatusFailError)
      ./features/step_definitions/submission_steps.rb:30:in `/^I am in (.*) and I guess my location on the submission page$/'
      features/nearby.feature:6:in `When I am in Winnipeg and I guess my location on the submission page'
    Then I should see an existing restroom nearby                        # features/step_definitions/restroom_steps.rb:9
```

> ` Request to 'http://127.0.0.1:36316/' failed to reach server, check DNS and/or server status - Timed out with the following resources still waiting http://www.google-analytics.com/analytics.js (Capybara::Poltergeist::StatusFailError)`

**Comment by DeeDeeG** [comment]
It passed this time. 👍 👍 👍 👍

**Comment by DeeDeeG** [comment]
Squash-and-merged this on the command-line:

`git merge --squash new-translation-ready develop`

So this PR is effectively merged in the following commit: https://github.com/RefugeRestrooms/refugerestrooms/commit/14e224173b4fa7b8c9b42c8cb5c17fc16d9ad722

Thanks everyone for the reviews!

**Pull request closed without merging**